### PR TITLE
Reset styles when popper is calculating position

### LINKED ISSUE #32484: Dropdown menu not positioned correctly with .dropstart
When using `.dropstart`, the dropdown menu will cover most of the toggle button.  

Seems like this is caused by this style:
https://github.com/twbs/bootstrap/blob/042a4a255f3c994ee8285df9a14fe0fbd95df83c/scss/_dropdown.scss#L36-L39

If I disable this style, the menu is positioned correctly.

This can be reproduced on the docs page:
https://getbootstrap.com/docs/5.0/components/dropdowns/#dropleft

- Operating system and version: Windows
- Browser and version: All

### PR DESCRIPTION
**Pull request by MartijnCuppens**
Farewell style hack 👋

Fixes https://github.com/twbs/bootstrap/issues/32484

https://deploy-preview-32510--twbs-bootstrap.netlify.app/docs/5.0/components/dropdowns/#dropleft

### CODE REVIEW

File: js/src/dropdown.js [hunk 1]
```diff
@@ -174,7 +174,9 @@ class Dropdown extends BaseComponent {
         }
       }
 
+      this._menu.classList.add('popper-active')
```
Comment (XhmikosR) [inline js/src/dropdown.js:177]: Can you please add a comment here why we add `popper-active` and we remove it after creating the popper?

Comment (MartijnCuppens) [reply js/src/dropdown.js]: Thanks for this, we don't need to remove this class indeed. I've added a comment about why we need to add this class.

Comment (MartijnCuppens) [reply js/src/dropdown.js]: Lol, ok, nevermind, we do need this to fix the dropleft positioning 😛

**Comment by ffoodd** [comment]
I'm not sure why but the behaviour changed in RTL, `translate()` does not move horizontally anymore: 
- This PR: https://deploy-preview-32510--twbs-bootstrap.netlify.app/docs/5.0/examples/cheatsheet-rtl/#dropdowns
- `main`: https://main--twbs-bootstrap.netlify.app/docs/5.0/examples/cheatsheet-rtl/#dropdowns

I can't see the relation with this PR though, but noticed it. Might be unrelated.

File: js/src/dropdown.js [hunk 2]
```diff
@@ -174,7 +175,12 @@ class Dropdown extends BaseComponent {
         }
       }
 
+      // The `.popper-active` class removes the `right` property in CSS
+      // This way popper can position the dropdown correctly
+      // We add it back after the calculations to get control over our positioning with our CSS again
+      this._menu.classList.add(CLASS_NAME_POPPER_ACTIVE)
       this._popper = Popper.createPopper(referenceElement, this._menu, this._getPopperConfig())
+      this._menu.classList.remove(CLASS_NAME_POPPER_ACTIVE)
```
Comment (septatrix) [inline js/src/dropdown.js:183]: Why do we remove the class again? When the element if positioned by popper the `right` property will not have an effect anyway or am I wrong?

Comment (MartijnCuppens) [reply js/src/dropdown.js:183]: That's what I would expect too, but it turns out this is needed by popper.

**Comment by rohit2sharma95** [comment]
As per the comment written in the code here:
https://github.com/twbs/bootstrap/blob/122b2bdd485373c05eeab50c9f809db31117b898/scss/_dropdown.scss#L36-L40

This right style was added to reset the position when the dropdown is positioned with Popper. (This was not in V4 where Popper V1 is used)
I tried to create a raw dropdown only with native Popper. And found that I don't need to reset the position when the element is placed by Popper.

As per my research, if you [remove the right style](https://github.com/rohit2sharma95/bootstrap/pull/14) from the dropdown, it works fine (even when Popper is not used): https://bootstrap-1jtqi8bem.vercel.app/docs/5.0/components/dropdowns/

Or maybe I am missing something @twbs/css-review :slightly_smiling_face:

**Comment by septatrix** [comment]
I tried to get rid of all the `top/right/bottom/left` styles which worker rather well except for the statically positioned examples (which do not use poppers dynamic positioning. It should work to only set these properties when `[data-bs-display=static]` matches or the element is inside a navbar (which are treated similar) however I do not have the time now to further experiment with this. Temporarily toggling the styles for popper however seems rather hacky (albeit appearing to work).

**Comment by rohit2sharma95** [comment]
Popper does not require the elements to have positioning styles and it can not place elements correctly that have positioning styles already.
https://codepen.io/rohit2sharma95/pen/wvzyGRM

> It's not really something Popper aims to support, **the requirement is that the target popper element should not be positioned already**.

https://github.com/popperjs/popper-core/discussions/1206#discussioncomment-252521

**Comment by mdo** [comment]
Is this supposed to fix the `dropdown-menu-end` alignment at https://deploy-preview-32510--twbs-bootstrap.netlify.app/docs/5.0/components/dropdowns/#menu-alignment? I still see the broken menu...

**Comment by rohit2sharma95** [comment]
> I still see the broken menu...

Because of the positioning of the dropdown initially.

**Comment by XhmikosR** [comment]
@rohit2sharma95 the issue mdo described above still happens. Was this supposed to be fixed by #32524?

**Comment by rohit2sharma95** [comment]
#32524 was just to remove the margin from the dropdown. The wrong placement of `dropdown-menu-end` is because of the `right` style IMO

**Comment by mdo** [comment]
Opened #32986 as an alternative for now.

**Comment by mdo** [comment]
Closing for #32986.

**Pull request closed without merging**